### PR TITLE
SonarQube refactoring Sprint 6

### LINKED ISSUE #226: [Refactoring] SonarQube Analysis report and refactoring - S6
As a developer, I would like to have clean code.

Acceptance Criteria:
- [x] Document and print the SonarQube Analysis run on Sprint 6
- [x] Refactor the found issues:
- [x]  Bug tolerance: 1
- [x]  Security tolerance: 1
- [x]  Code smell tolerance: 20
- [x]  Code dupplication tolerance: 3-4%

### PR DESCRIPTION
**Pull request by JayChouteau**
This is the refactoring of all issues based on SonarQube Analysis at the end of Sprint 6 (#226). Below are a few comments about the refactoring, some changes and issues faced:

1) Major changes
- All controller files were moved to a named package
- All UI utility files were moved to a named package
- The FXML has not been touched, the files simply were updated with the packages

2) Bug fixing
- In the "Server" pom file, the <scope> tag was removed for weka deep learning as well as the <systemPath> tag since the jar is installed for our project and only needs a regular dependency -> increasing portability

3) Code smells
- Blocker: missing assertions on system tests -> these will be addressed in the testing refactoring issues
- Critical: fixes were defining a constant instead of duplicating a literal
- Major: SonarQube suggests we used a logger instead of doing system. out.print. I left it for now and we could look at implementing it later. The fixes included removing unused or useless code as well as switching .isLast method to .next since .isLast it can be an expensive call and could pose problems.
- Minor: Notable fixes included changing access modifiers as well as using List instead of ArrayList in method signature for the following reason: "implementing classes must be used to instantiate new collections, but the result of an instantiation should ideally be stored in a variable whose type is a Java Collection interface."

### CODE REVIEW

File: JavaFXUserInterface/src/main/java/Controllers/AddSystemController.java [hunk 1]
```diff
@@ -17,6 +19,11 @@
 
 public class AddSystemController implements Initializable {
 
+    private static final String SAVE_DIALOG = "Save Dialog";
+    private static final String SAVE_HEADER = "Asset has been saved to the database.";
+    private static final String ERROR_DIALOG = "Error Dialog";
+    private static final String ERROR_HEADER = "Please enter values for all text fields.";
+    private static final String SYSTEMS = "/Systems";
```
Comment (PaulMicu) [inline JavaFXUserInterface/src/main/java/Controllers/AddSystemController.java:26]: if we want to do this, then it should be moved to the TextConstant file or another constant file

File: JavaFXUserInterface/src/main/java/Controllers/AddSystemTypeController.java [hunk 2]
```diff
@@ -6,14 +10,14 @@
...
 import javafx.scene.control.TextFormatter;
-import javafx.scene.layout.AnchorPane;
 import local.AssetTypeDAOImpl;
 
 import java.net.URL;
 import java.util.ArrayList;
 import java.util.ResourceBundle;
 
 public class AddSystemTypeController implements Initializable {
+    private final String SYSTEM_TYPE_LIST = "/Utilities.SystemTypeList";
```
Comment (PaulMicu) [inline JavaFXUserInterface/src/main/java/Controllers/AddSystemTypeController.java:20]: same as my other comment

File: JavaFXUserInterface/src/main/java/Controllers/SystemInfoController.java [hunk 3]
```diff
@@ -237,39 +230,39 @@ void deleteDialog(MouseEvent mouseEvent) {
...
         table.getItems().clear();
         Timeline timeline = new Timeline(new KeyFrame(Duration.millis(3000), e -> {
             ObservableList<AssetAttribute> attributes = FXCollections.observableArrayList(assetDAOImpl.createAssetInfo(system.getId()).getAssetAttributes());
 
             int columnIndex = 1;
-            TableColumn[] tableColumns = new TableColumn[attributes.size() + 1];
+            TableColumn<ObservableList<String>, String>[] tableColumns = new TableColumn[attributes.size() + 1];
 
-            ArrayList<ArrayList<Measurement>> data = new ArrayList();
+            ArrayList<List<Measurement>> data = new ArrayList<>();
```
Comment (PaulMicu) [inline JavaFXUserInterface/src/main/java/Controllers/SystemInfoController.java:241]: why change the arraylist to a list?

File: JavaFXUserInterface/src/main/java/Controllers/SystemTypeController.java [hunk 4]
```diff
@@ -61,11 +78,10 @@ public void fillSystemTypeTable() {
      * This method will return an ObservableList of SystemList objects
      *
      * @author Shirwa
-     *
+     * <p>
```
Comment (PaulMicu) [inline JavaFXUserInterface/src/main/java/Controllers/SystemTypeController.java]: why?

File: JavaFXUserInterface/src/main/java/Controllers/SystemTypeInfoController.java [hunk 5]
```diff
@@ -14,6 +18,7 @@
 import java.util.ResourceBundle;
 
 public class SystemTypeInfoController implements Initializable {
+    private final String SYSTEM_TYPE_LIST = "/Utilities.SystemTypeList";
```
Comment (PaulMicu) [inline JavaFXUserInterface/src/main/java/Controllers/SystemTypeInfoController.java:21]: same as my first comment

File: JavaFXUserInterface/src/main/java/Controllers/SystemTypeInfoController.java [hunk 6]
```diff
@@ -83,69 +102,68 @@ void initData(SystemTypeList assetType) {
      * Attaches events to elements in the scene.
      *
      * @author Najim
-     *
+     * <p>
```
Comment (PaulMicu) [inline JavaFXUserInterface/src/main/java/Controllers/SystemTypeInfoController.java]: again why?

File: JavaFXUserInterface/src/main/java/Utilities/UIUtilities.java [hunk 7]
```diff
@@ -17,12 +20,49 @@
...
+     */
+    public static TextFormatter.Change checkFormat(DecimalFormat format, TextFormatter.Change c) {
+        if (c.getControlNewText().isEmpty())
+            return c;
+        ParsePosition parsePosition = new ParsePosition(0);
+        if (format.parse(c.getControlNewText(), parsePosition) == null || parsePosition.getIndex() < c.getControlNewText().length())
+            return null;
+        return c;
+    }
+
```
Comment (PaulMicu) [inline JavaFXUserInterface/src/main/java/Utilities/UIUtilities.java:60]: why move this function a 100 lines up...

File: Server/src/main/java/preprocessing/DataPreProcessorImpl.java [hunk 8]
```diff
@@ -90,21 +91,68 @@ private static Instances addRUL(Instances trainingData, ArrayList<Double> maxCyc
...
+
+    private static boolean setContains(Instances dataset, Attribute att) {
+        for (int i = 0; i < dataset.numAttributes(); i++) {
+            if (att.name().equals(dataset.attribute(i).name())) {
+                return true;
+            }
+
+        }
+        return false;
+    }
```
Comment (PaulMicu) [inline Server/src/main/java/preprocessing/DataPreProcessorImpl.java:155]: Same as other comment, why move this function

**Review by PaulMicu — CHANGES_REQUESTED** [review]
A few questions, please see line comments

File: JavaFXUserInterface/src/main/java/Controllers/AddSystemController.java (hunk 1 above)
Comment (JayChouteau) [reply JavaFXUserInterface/src/main/java/Controllers/AddSystemController.java:26]: I wanted to leave it for the code convention enhancement issue. But if I am gonna touch on it, I can pretty much do it at this point, I'll add it.

File: JavaFXUserInterface/src/main/java/Controllers/SystemInfoController.java (hunk 3 above)
Comment (JayChouteau) [reply JavaFXUserInterface/src/main/java/Controllers/SystemInfoController.java:241]: The reason is the following: implementing classes must be used to instantiate new collections, but the result of an instantiation should ideally be stored in a variable whose type is a Java Collection interface

File: JavaFXUserInterface/src/main/java/Controllers/SystemTypeController.java (hunk 4 above)
Comment (JayChouteau) [reply JavaFXUserInterface/src/main/java/Controllers/SystemTypeController.java]: sorry, no idea what that is, i'll remove it.

File: JavaFXUserInterface/src/main/java/Utilities/UIUtilities.java (hunk 7 above)
Comment (JayChouteau) [reply JavaFXUserInterface/src/main/java/Utilities/UIUtilities.java:60]: I didn't manually change it. I am guessing it did automatically when I did the code cleanup by IntelliJ. It automatically moves static functions at the top of the file.

File: JavaFXUserInterface/src/main/java/Controllers/SystemInfoController.java (hunk 3 above)
Comment (PaulMicu) [reply JavaFXUserInterface/src/main/java/Controllers/SystemInfoController.java:241]: ok

File: JavaFXUserInterface/src/main/java/Controllers/AddSystemController.java (hunk 1 above)
Comment (PaulMicu) [reply JavaFXUserInterface/src/main/java/Controllers/AddSystemController.java:26]: ok, just leave if for that issue then

**Comment by JayChouteau** [comment]
@PaulMicu the <p> comment was fixed if you want to approve and merge

**Review by PaulMicu — APPROVED** [review]

**Pull request merged**